AutomaticQC/imosEchoRangeSetQC.m

```diff
@@ -0,0 +1,167 @@
+% it under the terms of the GNU General Public License as published by
+% the Free Software Foundation version 3 of the License.
+%
+% This program is distributed in the hope that it will be useful,
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+% GNU General Public License for more details.
+
+% You should have received a copy of the GNU General Public License
+% along with this program.
+% If not, see <https://www.gnu.org/licenses/gpl-3.0.en.html>.
+%
+narginchk(1, 2);
+if ~isstruct(sample_data), error('sample_data must be a struct'); end
+
+% auto logical in input to enable running under batch processing
+if nargin<2, auto=false; end
+
+varChecked = {};
+paramsLog  = [];
+
+% get all necessary dimensions and variables id in sample_data struct
+idUcur = 0;
+idVcur = 0;
+idWcur = 0;
+idCspd = 0;
+idCdir = 0;
+idABSIC = cell(4, 1);
+for j=1:4
+    idABSIC{j}  = 0;
+end
+lenVar = length(sample_data.variables);
+for i=1:lenVar
+    paramName = sample_data.variables{i}.name;
+    
+    if strncmpi(paramName, 'UCUR', 4),  idUcur = i; end
+    if strncmpi(paramName, 'VCUR', 4),  idVcur = i; end
+    if strcmpi(paramName, 'WCUR'),      idWcur = i; end
+    if strcmpi(paramName, 'CSPD'),      idCspd = i; end
+    if strncmpi(paramName, 'CDIR', 4),  idCdir = i; end
+    for j=1:4
+        cc = int2str(j);
+        if strcmpi(paramName, ['ABSIC' cc]), idABSIC{j} = i; end
+    end
+end
+
+% check if the data is compatible with the QC algorithm
+idMandatory = (idUcur | idVcur | idWcur | idCspd | idCdir);
+for j=1:4
+    idMandatory = idMandatory & idABSIC{j};
+end
+if ~idMandatory, return; end
+
+% let's get the associated vertical dimension
+idVertDim = sample_data.variables{idABSIC{1}}.dimensions(2);
+if strcmpi(sample_data.dimensions{idVertDim}.name, 'DIST_ALONG_BEAMS')
+    disp(['Warning : imosEchoRangeSetQC applied with a non tilt-corrected ABSICn (no bin mapping) on dataset ' sample_data.toolbox_input_file]);
+end
+
+qcSet           = str2double(readProperty('toolbox.qc_set'));
+badFlag         = imosQCFlag('bad',             qcSet, 'flag');
+goodFlag        = imosQCFlag('good',            qcSet, 'flag');
+rawFlag         = imosQCFlag('raw',             qcSet, 'flag');
+
+%Pull out echo intensity
+sizeData = size(sample_data.variables{idABSIC{1}}.data);
+ea = nan(4, sizeData(1), sizeData(2));
+for j=1:4
+    ea(j, :, :) = sample_data.variables{idABSIC{j}}.data;
+end
+
+% read in filter parameters
+propFile  = fullfile('AutomaticQC', 'imosEchoRangeSetQC.txt');
+ea_fishthresh = str2double(readProperty('ea_fishthresh',   propFile));
+
+% read dataset QC parameters if exist and override previous 
+% parameters file
+currentQCtest = mfilename;
+ea_fishthresh = readDatasetParameter(sample_data.toolbox_input_file, currentQCtest, 'ea_fishthresh', ea_fishthresh);
+
+paramsLog = ['ea_fishthresh=' num2str(ea_fishthresh)];
+
```

Comment:
need to follow parramsLog "standards": use parenthesis, arguments used, the outcomes of the test, etc.